### PR TITLE
Add role certificate validity check when loading manifest in core

### LINKED ISSUE #727: Core should verify the chain of signatures to validate the role of the user (writer, owner etc.)
Currently, the verification of the user roles is performed only by the back-end. The same verification should be performed by users after receiving content from metadata server.

This mechanism is similar with validating administrative signed operations, which happen both at the back-end and at the users side.

### PR DESCRIPTION
**Pull request by touilleMan**
fix #727

### CODE REVIEW

File: tests/core/fs/test_fs_sync.py [hunk 1]
```diff
@@ -79,7 +79,8 @@
 @pytest.mark.trio
 @pytest.mark.parametrize("type", ["file", "folder"])
 async def test_new_empty_entry(type, running_backend, alice_user_fs, alice2_user_fs):
-    wid = await create_shared_workspace("w", alice_user_fs, alice2_user_fs)
+    with freeze_time("2000-01-01"):
+        wid = await create_shared_workspace("w", alice_user_fs, alice2_user_fs)
```
Comment (vxgmichel) [inline tests/core/fs/test_fs_sync.py]: Why is that necessary?

File: parsec/core/fs/remote_loader.py [hunk 2]
```diff
@@ -325,13 +343,26 @@ def __init__(
...
+        )
+        if role_at_timestamp is None:
+            raise FSError(
+                f"Manifest was created at {expected_timestamp} by `{expected_author}` "
+                "which had no right to access the workspace at that time"
+            )
+        elif role_at_timestamp == RealmRole.READER:
+            raise FSError(
+                f"Manifest was created at {expected_timestamp} by `{expected_author}` "
+                "which had write right on the workspace at that time"
```
Comment (vxgmichel) [inline parsec/core/fs/remote_loader.py]: `which had no write right`? `which had no right to write in the workspace`?

File: tests/core/fs/workspacefs/test_invalid_data.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+            # This should trigger FSError
+            with pytest.raises(FSError) as exc:
+                await workspace.sync()
+            str(exc.value) == exc_msg
+
+            # Also test timestamped workspace
+            with pytest.raises(FSError) as exc:
+                await workspace.to_timestamped(options["backend_timestamp"])
+            str(exc.value) == exc_msg
```
Comment (vxgmichel) [inline tests/core/fs/workspacefs/test_invalid_data.py]: Maybe parametrize the `testbed` fixture and let the test catch the `FSError` exception? e.g:
```python
@pytest.mark.trio
async def test_empty_blob(testbed):
    with pytest.raises(FSError) as context:
        await testbed.run(blob=b"")
    assert context.value.message == "Cannot decrypt vlob: The nonce must be exactly 24 bytes long"
```

File: tests/core/fs/test_fs_sync.py (hunk 1 above)
Comment (touilleMan) [reply tests/core/fs/test_fs_sync.py]: Because otherwise shared workspace is created with current time, but after that we do a `workspace.sync()` with time frozen at 2000-01-03.
So the timestamp consistency test failed given there is no role certificate available at the time we upload the data

File: parsec/core/fs/remote_loader.py (hunk 2 above)
Comment (touilleMan) [reply parsec/core/fs/remote_loader.py]: wrong wording or poor style ?

File: tests/core/fs/workspacefs/test_invalid_data.py (hunk 3 above)
Comment (touilleMan) [reply tests/core/fs/workspacefs/test_invalid_data.py]: That was exactly what I was doing before adding the timestamped part ^^

Before that we had only one call raising an exception so it was logical to let the caller test handle it
But with two errors being raised, I should have two methods (well 3 in fact: `init`, `run_workspace`, `run_timestamped_workspace`) that have to be called by the test with exception handling. This is much more cumbersome (especially considering the methods may not be easily idempotent) so I just KISS
